modules/javafx.controls/src/main/java/javafx/scene/control/skin/MenuBarSkin.java

```diff
@@ -474,10 +472,10 @@ private void showMenu(Menu menu, boolean selectFirstItem) {
     }
 
     private void setFocusedMenuIndex(int index) {
-        this.focusedMenuIndex = index;
-        focusedMenu = index == -1 ? null : getSkinnable().getMenus().get(index);
+        focusedMenuIndex = (index >= -1 && index < getSkinnable().getMenus().size())? index : -1;
+        focusedMenu = (focusedMenuIndex != -1)? getSkinnable().getMenus().get(index) : null;
```

Comment:
Minor: we usually put a space before the `?` (I probably wouldn't have mentioned it, but there is a substantive change you need to make anyway).